skbio/sequence/_sequence.py

```diff
@@ -373,6 +376,7 @@ def metadata(self):
         return self._metadata
 
     @metadata.setter
+    @stable(as_of="0.4.0")
```

Comment:
I agree, the getter/setter/deleter are linked to each other and we only have a docstring on the getter, so the getter should be the only one decorated.